app/controllers/items_controller.rb

```diff
@@ -68,14 +69,14 @@ def create
   end
   
   def search
-    if !params[:search].empty?
+    if params[:search] == nil || params[:search].empty?
```

Comment:
if `params[:search].nil?`

Reply:
changed button_to to link_to so the clear search functionality doesn't touch the controller anymore (just links to root_path)